File: Classes/WMISet/WMISet/Examples/delphi/ProcessControl/FrmNewTaskU.pas
unit FrmNewTaskU;

interface
{$I define.inc}

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls;

type
  TFrmNewTask = class(TForm)
    lblFileName: TLabel;
    edtFileName: TEdit;
    btnStart: TButton;
    btnCancel: TButton;
    procedure edtFileNameChange(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  FrmNewTask: TFrmNewTask;

implementation

{$R *.dfm}

procedure TFrmNewTask.edtFileNameChange(Sender: TObject);
begin
  btnStart.Enabled := Trim(edtFileName.Text) <> '';
end;

end.

File: Classes/WMISet/WMISet/Examples/cbuilder/ProcessControl/FrmNewTaskU.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "FrmNewTaskU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TFrmNewTask *FrmNewTask;
//---------------------------------------------------------------------------
__fastcall TFrmNewTask::TFrmNewTask(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TFrmNewTask::edtFileNameChange(TObject *Sender)
{
  btnStart->Enabled = edtFileName->Text.Trim().Length() > 0;
}
//---------------------------------------------------------------------------
